p2p/net/conn/listen.go

```diff
@@ -157,12 +135,62 @@ func (l *listener) Loggable() map[string]interface{} {
 	}
 }
 
+func (l *listener) handleIncoming() {
+	var wg sync.WaitGroup
+	defer func() {
+		wg.Wait()
+		close(l.incoming)
+	}()
+
```

Comment:
it needed to be shared across two different goroutines, i moved it into the listener struct